modal/_container_entrypoint.py

```diff
@@ -739,6 +751,12 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
 
         # Initialize the function, importing user code.
         with container_io_manager.handle_user_exception():
+            if max_concurrency != 0 and max_concurrency <= target_concurrency:
```

Comment:
We don't need these checks - I think it's odd for a function to throw this when starting up, if it could have been caught earlier, and we can just allow it to start up